submissions/KostiantynOkhorzin/memory-game/script.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const gameBody = document.querySelector('.game__body');
+let playerSteps = 0;
+const cardProcessingTime = 1500;
+
+const dataImgs = [
+    { imgSrc: "img/ananas.png", name: "ananas" },
+    { imgSrc: "img/apple.png", name: "apple" },
+    { imgSrc: "img/avocado.png", name: "avocado" },
+    { imgSrc: "img/banana.png", name: "banana" },
+    { imgSrc: "img/kiwi.png", name: "kiwi" },
+    { imgSrc: "img/lime.png", name: "lime" },
+    { imgSrc: "img/ananas.png", name: "ananas" },
+    { imgSrc: "img/apple.png", name: "apple" },
+    { imgSrc: "img/avocado.png", name: "avocado" },
+    { imgSrc: "img/banana.png", name: "banana" },
+    { imgSrc: "img/kiwi.png", name: "kiwi" },
+    { imgSrc: "img/lime.png", name: "lime" },
+];
+
+const styleList = {
+    card: 'game__card',
+    flippedCard: 'game__card_flipped',
+    selectedCard: 'game__card_selected',
+    hiddenCard: 'game__card_hidden',
+    frontCard: 'game__card-front',
+    backCard: 'game__card-back'
+}
+
+class GameCard {
+    constructor(img, name) {
+        this.img = img;
+        this.name = name;
+    }
+
+    render() {
+        gameBody.insertAdjacentHTML('beforeend', `
+            <button class=${styleList.card} data-name=${this.name}>
+                <span class=${styleList.frontCard}></span>
+                <img class=${styleList.backCard} src=${this.img} alt=${this.name}>
+            </button>
+        `);
+    }
+};
+
+const randomlyMixArrayElem = () => {
+    return dataImgs.sort(() => Math.random() - 0.5);
+};
+
+const renderCards = () => {
+    randomlyMixArrayElem().map(({ imgSrc, name }) => new GameCard(imgSrc, name).render());
+};
+
+const handleCardFlip = ({ target }) => {
+    if (target.classList.contains(styleList.card)) {
+        target.classList.add(styleList.selectedCard, styleList.flippedCard);
+        const flippedCards = document.querySelectorAll(`.${styleList.flippedCard}`);
+        const selectedCards = document.querySelectorAll(`.${styleList.selectedCard}`);
+        if (selectedCards.length === 2) {
+            if (selectedCards[0].dataset.name === selectedCards[1].dataset.name) {
```

Comment:
This block (lines 61-74) has some repeating code. Try to figure out how you can shorten it.